### PR TITLE
chore: use nuxt kit

### PR DESCRIPTION
**Pull request by manniL**
# TODO

* [x] Set up Nuxt 2 playground
* [x] Set up Nuxt 3 playground
* [x] Verify changes work, e.g. by testing in https://github.com/albionstatus/albionstatus-website
* [x] Maybe even tests via `test-utils` to very google adsense component was rendered 😋

### CODE REVIEW

**Comment by manniL** [comment]
PR should be ready to review, cc @pi0 and @danielroe ☺️

File: src/module.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      nuxt: '^2.15.0 || ^3.0.0-rc.11'
+    }
+  },
+  defaults: (nuxt) => ({
+    ...DEFAULTS,
+    test: nuxt.options.dev && process.env.NODE_ENV !== 'production'
+  }),
+  setup (options, nuxt) {
+    if (options.test) {
+      logger.info('Test mode enabled - Using Test Adsense ID')
```
Comment (danielroe) [inline src/module.ts:22]: might be worth using consistent 'adsense' case throughout

File: src/module.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    }
+  }
+})
+
+function createScriptMeta (script: string, isNuxt2: boolean) {
+  // Ensure `window.adsbygoogle` is defined
+  script = `(window.adsbygoogle = window.adsbygoogle || []); ${script}`
+  // wrap script inside a guard check to ensure it executes only once
+  script = `if (!window.__abg_called){ ${script} window.__abg_called = true;}`
+  const key = isNuxt2 ? 'innerHTML' : 'children'
```
Comment (danielroe) [inline src/module.ts:118]: I think we support `innerHTML` for Nuxt 3 as well - in fact, it's now preferred.

File: src/module.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+          alias: options.tag
+        }
+      })
+    } else {
+      // Add component to auto load
+      addComponentsDir({
+        path: resolveRuntimeDir('components-v3'),
+        isAsync: false,
+        prefix: '',
+        level: 999
```
Comment (danielroe) [inline src/module.ts:94]: I think you can safely omit these.
```suggestion
```

**Review by danielroe — APPROVED** [review]
Looks good to me, though I've not tested it.

File: src/module.ts (hunk 1 above)
Comment (danielroe) [reply src/module.ts:22]: Just checked; I think it is [AdSense](https://www.google.com/intl/nl_nl/adsense/start/).

Comment (manniL) [reply src/module.ts:22]: Alright, will incorporate that one then 😋

Comment (danielroe) [reply src/module.ts:22]: sorry 💀

**Pull request merged**